### PR TITLE
export data to excel sheet functionality added

### PR DESCRIPTION
**Pull request by Ayushjaisgit**

### CODE REVIEW

File: controllers/task.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const { sendResponse } = require('../helpers/sendResponse');
 const { populate } = require('../models/ratings');
```
Comment (manavsharma13) [inline controllers/task.js:5]: Change this name according to model name

Comment (manavsharma13) [inline controllers/task.js]: This is wrong no populate function is present in models/ratings and line is having no use please remove this

File: controllers/task.js [hunk 2]
```diff
@@ -417,6 +418,126 @@ const getGroupByTasks = async (req, res, next) => {
...
+		data.aggregateGroupBy = `$${data.groupBy}`
+	}
+
+	data.aggregateGroupBy = CONSTANTS.GROUP_BY[data.groupBy]
+
+	let taskRes = await createPayloadAndGetGroupByTask(data)
+	let taskData = taskRes.data
+	// data.filteredData taskRes.data;
+	console.log(taskData.tasks)
+	if (taskRes.error) {
```
Comment (manavsharma13) [inline controllers/task.js]: just suggestion for code writing no need to  add space after if and curly bracket

File: routes/task.js [hunk 3]
```diff
@@ -13,7 +13,7 @@ const authenticateRole = require("../middlewares/authenticateRole");
 const filterProjects = require("../middlewares/filterProjectsForRoles")();
```
Comment (manavsharma13) [inline routes/task.js:13]: used var in sendResponse function in middleware

File: controllers/task.js [hunk 4]
```diff
@@ -417,6 +418,126 @@ const getGroupByTasks = async (req, res, next) => {
...
+
+			let assignedTo = ''
+			let assignedToUser = tasks.assignedTo?.name 
+			if (assignedToUser===null) {
+				assignedToUser = tasks.assignedTo;
+			}
+			assignedTo = assignedToUser;
+
+			let createdBy = ''
+			let createdByUser =  tasks.createdBy?.name
```
Comment (manavsharma13) [inline controllers/task.js]: let createdByUser =  tasks.createdBy?.name 
			if (createdByUser===null) {
				createdByUser = tasks.createdBy;
			}
if name is coming everytime in createdBy as we must having name in creadtedBy object then there is no use of second if condition

Comment (Ayushjaisgit) [reply controllers/task.js]: got it sir! I have removed if conditional

File: routes/task.js (hunk 3 above)
Comment (Ayushjaisgit) [reply routes/task.js:13]: changed to Let!

File: controllers/task.js (hunk 1 above)
Comment (Ayushjaisgit) [reply controllers/task.js]: corrected sir !

File: controllers/task.js (hunk 2 above)
Comment (Ayushjaisgit) [reply controllers/task.js]: Corrected it sir! Thank you for the suggestion

**Comment by Ayushjaisgit** [comment]
All Code review corrections have been made

**Pull request merged**